pypasta/Repository/Mbox.py

```diff
@@ -371,6 +375,102 @@ def __getitem__(self, message_id):
         return ret
 
 
+class PatchworkProject(MailContainer):
+    def __init__(self, project_id, d_mbox, f_index, f_mbox_raw, **kwargs):
+        self.patchwork_url = kwargs.get('url')
+        self.project_id = project_id
+        self.f_index = f_index
+        self.f_mbox_raw = f_mbox_raw
+        self.d_mbox = d_mbox
+        self.d_mbox_patchwork = os.path.join(self.d_mbox, 'patchwork')
+
+        self.index = self.load_index(self.f_index, is_patchwork_index=True)
+        log.info('  ↪ loaded mail index for Patchwork project id %u: found %d mails' %
+                 (project_id, len(self.index)))
+
+    @staticmethod
+    def page_iterator(resp, yield_value, since):
+        done = False
+        while True:
+            if resp.status_code != 200:
+                break
+            for item in resp.json():
+                value = yield_value(item)
+                if value:
+                    if since is not None:
+                        if str(value[-1]) == since:
+                            done = True
+                            break
+                    yield value
+            if done:
+                break
+            next_page = resp.links.get('next', None)
+            if next_page is None:
+                break
+            resp = requests.get(next_page['url'])
+
+    def download_patches(self, since=None):
+        def yield_value(patch):
+            patchwork_id = patch['id']
+            date = datetime.fromisoformat(patch['date'])
+            msg_id = patch['msgid']
+            mbox_url = patch['mbox']
+            resp_mbox = requests.get(mbox_url)
+
+            return date, msg_id, resp_mbox.text, patchwork_id
+
+        log.info('Downloading patches from Patchwork for project id %u' % self.project_id)
+        params = {}
+        params['order'] = '-id'   # descending id order
+        params['project'] = self.project_id
+        resp = requests.get(urljoin(self.patchwork_url, 'patches'), params)
+        return self.page_iterator(resp, yield_value, since)
+
+    def update(self):
+        if self.f_mbox_raw != "":
+            log.info('Processing raw mailbox for Patchwork project id %u' %
+                     self.project_id)
+            load_mbox_file(self.f_mbox_raw, str(self.project_id),
+                           self.d_mbox, mode='patchwork')
```

Comment:
Add a report here that the initial import was successful, and guide the user to remove the initial archive configuration option. Return here.